HW3/P4/median_filter.cl

```diff
@@ -13,7 +85,27 @@ median_3x3(__global __read_only float *in_values,
     // without using the local buffer, first, then adjust your code to
     // use such a buffer after you have that working.
 
+    //store calculated median here before transferring 
+    float out_buffer; 
+    
+    // Global position of output pixel
+    const int x = get_global_id(0);
+    const int y = get_global_id(1);
 
+    // Local position relative to (0, 0) in workgroup
+    const int lx = get_local_id(0);
+    const int ly = get_local_id(1);
+
+    // coordinates of the upper left corner of the buffer in image
+    // space, including halo
+    const int buf_corner_x = x - lx - halo;
+    const int buf_corner_y = y - ly - halo;
+
+    const int idx_1D = ly * get_local_size(0) + lx;
+
+    int row;
+
+    if (x < w && y < h){
```

Comment:
Adding this line here may cause a partial population of the buffer (for example, without the column that is to the right of the right-most column), but then when reading from the buffer (lines 135-143), you may read an invalid value.